src/dvc_data/hashfile/build.py

```diff
@@ -252,8 +232,6 @@ def build(
             **kwargs,
         )
         logger.debug("built tree '%s'", obj)
-        if name != "md5":
-            obj = _build_external_tree_info(odb, obj, name)
```

Comment:
After the change in https://github.com/iterative/dvc/pull/8804, I noticed that the `obj` returned here is just identical to the one returned by `_build_tree` above 🤷 

Is there something I am missing @efiop ?

Reply:
Not sure how I missed this, sorry. It might be etag-based on s3, IIRC.